### PR TITLE
Remove credential

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

**Comment by mattcollier** [comment]
Couple more additions coming...

File: lib/store.js [hunk 1]
```diff
@@ -436,3 +438,64 @@ Store.prototype.compose = function(
     callback(err, identity);
   });
 };
+
+/**
+ * Marks a Credential as deleted.
+ *
+ * @param actor the Identity performing the action.
+ * @param credentialId the Credential ID.
+ * @param status the status.
```
Comment (dlongley) [inline lib/store.js]: Erroneous `@param`.

File: lib/store.js [hunk 2]
```diff
@@ -436,3 +438,64 @@ Store.prototype.compose = function(
...
+ * @param callback(err) called once the operation completes.
+ */
+Store.prototype.remove = function(actor, credentialId, callback) {
+  var self = this;
+  async.auto({
+    find: function(callback) {
+      self.collection.findOne({
+        'credential.id': credentialId
```
Comment (dlongley) [inline lib/store.js]: Change to use indexed id.

File: lib/store.js [hunk 3]
```diff
@@ -436,3 +438,64 @@ Store.prototype.compose = function(
...
+ *
+ * @param actor the Identity performing the action.
+ * @param credentialId the Credential ID.
+ * @param status the status.
+ * @param callback(err) called once the operation completes.
+ */
+Store.prototype.remove = function(actor, credentialId, callback) {
+  var self = this;
+  async.auto({
+    find: function(callback) {
```
Comment (dlongley) [inline lib/store.js]: Do we typically run `find` in a remove call? I suppose you'd get a permission denied error instead if we didn't...

File: lib/store.js [hunk 4]
```diff
@@ -436,3 +438,64 @@ Store.prototype.compose = function(
...
+          ));
+        }
+        callback(null, result);
+      });
+    },
+    checkPermission: ['find', function(callback, results) {
+      var credential = results.find.credential;
+      brPermission.checkPermission(
+        actor, PERMISSIONS.CREDENTIAL_REMOVE,
+        {resource:credential, translate: 'claim'}, callback);
```
Comment (dlongley) [inline lib/store.js]: Missing space: `resource:credential`

File: lib/store.js [hunk 5]
```diff
@@ -436,3 +438,63 @@ Store.prototype.compose = function(
...
+  var self = this;
+  async.auto({
+    find: function(callback) {
+      self.collection.findOne({
+        id: database.hash(credentialId)
+      }, function(err, result) {
+        if(err) {
+          return callback(err);
+        }
+        if(!('credential' in result)) {
```
Comment (dlongley) [inline lib/store.js]: Isn't `result` null when there's no result?

**Comment by dlongley** [comment]
One more comment to address, then LGTM, merge.

**Pull request merged**